r-package/grf/vignettes/diagnostics.Rmd

```diff
@@ -31,22 +34,67 @@ Y <- pmax(X[, 1], 0) * W + X[, 2] + pmin(X[, 3], 0) + rnorm(n)
 cf <- causal_forest(X, Y, W)
 ```
 
-The overlap assumptions requires a positive probability of treatment for each $X_i$. We should not be able to deterministically decide the treatment status of an individual based on its covariates, meaning none of the estimated propensity scores should be close to one or zero. One can check this with a histogram:
+The overlap assumption requires a positive probability of treatment for each $X_i$. We should not be able to deterministically decide the treatment status of an individual based on its covariates, meaning none of the estimated propensity scores should be close to one or zero. One can check this with a histogram:
 
 ```{r}
 hist(e.hat <- cf$W.hat)
 ```
 
-One can also check that the covariates are balanced across the control and treated group by plotting the histogram $X_i | W_i = k$, overlaid here for each feature:
+One can also check that the covariates are balanced across the treated and control group by plotting the inverse-propensity weighted histograms of all samples, overlaid here for each feature (done with [ggplot2](https://ggplot2.tidyverse.org/index.html) which supports weighted histograms):
+
+```{r, fig.height = 10}
+plot.df <- data.frame(X, W = as.factor(W), IPW = ifelse(W == 1, 1 / e.hat, 1 / (1 - e.hat)))
+plot.df <- reshape(plot.df, varying = list(1:p), v.names = "X", direction = "long")
```

Comment:
It'd be nice if this command could preserve the feature names. Now it just shows feature numbers 1, 2, ..., p.